pkg/util/utils.go

```diff
@@ -96,8 +95,20 @@ func RetrievePlatformInfoFromConfig(config *rest.Config, params map[string]inter
 		return errors.Errorf("Failed to get k8s secret, %s", vsphere_secrets)
 	}
 
-	sEnc := string(secret.Data[secretData])
-	lines := strings.Split(sEnc, "\n")
+	// No kv pairs in the secret.
+	if len(secret.Data) == 0 {
+		errMsg := fmt.Sprintf("Failed to get k8s secret, %s", vsphere_secrets)
+		return errors.New(errMsg)
+	}
+
+	var sEnc string
+	var lines []string
+	// Possibility of multiple kv pairs
+	for _, value := range secret.Data {
```

Comment:
I don't understand this code.  Why are we doing a for but then only using the first value?

Reply:
The expectation is the secret.Data has only one kv pair(https://vsphere-csi-driver.sigs.k8s.io/driver-deployment/installation.html), the key being the name of the file that was used to create the Secret, and the value being the byte representation of the actual VC configuration.